pkg/apis/planetscale/v2/vitessshard_defaults.go

```diff
@@ -16,10 +16,24 @@ limitations under the License.
 
 package v2
 
+import (
+	"k8s.io/utils/pointer"
+)
+
 // DefaultVitessShard fills in VitessShard defaults for unspecified fields.
 // Note: This should only be used for nillable fields passed down from a parent because controllers run in parallel,
 // and the defaulting code for a parent object may not have been run yet, meaning the values passed down from that parent
 // might not be safe to deref.
 func DefaultVitessShard(dst *VitessShard) {
 	DefaultTopoReconcileConfig(&dst.Spec.TopologyReconciliation)
+	defaultInitializeMaster(&dst.Spec.VitessShardTemplate.Replication.InitializeMaster)
 }
+
+func defaultInitializeMaster(enabled *bool) {
```

Comment:
I'm a little confused. Wouldn't this be a `**bool` at this point because `InitializeMaster` is a pointer bool already, so we need to pass the address to that pointer bool?

Reply:
The only time it will be called, it will be defaulted, so we just call the address of that defaulted template.

Reply:
I see my confusion - you changed it to a normal bool. I think it should be a pointer bool so we can change the default over time. This also has been our general pattern for code that we default (making them pointer types).

Reply:
Peter is right; this would have to be a `**bool`.

However, I recommend changing the function to be `DefaultVitessReplicationSpec(*VitessReplicationSpec)` because we don't usually create a whole defaulter function for just one bool.

We should also then call `DefaultVitessReplicationSpec()` on every `VitessShardTemplate` nested within a `VitessCluster`. This is the first time we've had a non-zero default in that struct, so you'll need to create a few of the intermediate methods. We currently only go as deep as `DefaultVitessKeyspaceTemplate()`.